File: VerbumKit/Sources/Clients/AccountClient.swift
import ComposableArchitecture
import Models

@DependencyClient
public struct AccountClient: Sendable {
    public var sessions: @Sendable () -> AsyncStream<AuthSession?> = { .finished }
    public var signIn: @Sendable (_ email: String, _ password: String) async throws -> AuthSession
    public var register: @Sendable (_ email: String, _ password: String) async throws -> AuthSession
    public var anonymous: @Sendable () async throws -> AuthSession
    public var resetPassword: @Sendable (_ email: String) async throws -> Void
    public var sendVerification: @Sendable () async throws -> Void
    public var refresh: @Sendable () async throws -> AuthSession?
    public var signOut: @Sendable () async throws -> Void
    public var deleteAccount: @Sendable (_ password: String) async throws -> Void
}

extension AccountClient: DependencyKey {
    public static let liveValue = Self.firebase
    public static let previewValue = Self(sessions: { AsyncStream { $0.yield(nil); $0.finish() } })
}
extension DependencyValues {
    public var accountClient: AccountClient {
        get { self[AccountClient.self] }
        set { self[AccountClient.self] = newValue }
    }
}

File: android/core/clients/src/main/kotlin/com/nexussoft/verbum/clients/AccountClient.kt
package com.nexussoft.verbum.clients

import com.nexussoft.verbum.models.AuthSession
import kotlinx.coroutines.flow.Flow

/** Credentials belong to the SDK, never preferences or saved UI state. */
interface AccountClient {
    fun sessions(): Flow<AuthSession?>
    suspend fun signIn(email: String, password: String): AuthSession
    suspend fun register(email: String, password: String): AuthSession
    suspend fun anonymous(): AuthSession
    suspend fun resetPassword(email: String)
    suspend fun sendVerification()
    suspend fun refresh(): AuthSession?
    suspend fun signOut()
    suspend fun deleteAccount(password: String)
}
